src/cocotb_bus/scoreboard.py

```diff
@@ -71,7 +71,7 @@ def result(self):
                 fail = True
         assert not fail, "Not all expected output was received"
         assert not self.errors, "Errors were recorded during the test"
-        return TestSuccess()
+        return test_success()
```

Comment:
`cocotb.pass_test` does not return a `TestSuccess`, but immediately ends the test.
```suggestion
        test_success()
```
This is an interface breaking change and we will need to bump the version of cocotb-bus, but that's okay and expected. I seriously doubt anyone is using `Scoreboard`. It's a goddamn mess.

Reply:
Apologies for not getting around to making the final changes quick enough. I'm curious what you would recommend for a Scoreboard? I have been using cocotb-bus Scoreboard so If there is something better I'd be happy to use it.

Reply:
I actually kept the code you wrote instead of what's here.

I'm surprised it works for you. It had some very weird behavior were couldn't reconcile when str or bytes were put into it. Maybe it's fine for other types.